src/index.js

```diff
@@ -41,6 +41,40 @@ const _H = 'H'.charCodeAt(0);
 const _Y = 'Y'.charCodeAt(0);
 const _S = 's'.charCodeAt(0);
 
+function isBufferPNG(buffer) {
+  if (buffer.length < 8) return;
+  return buffer[0] === 137 && buffer[1] === 80 && buffer[2] === 78 && buffer[3] === 71 && buffer[4] === 13 && buffer[5] === 10 && buffer[6] === 26 && buffer[7] === 10;
+}
+
+function isBufferJPEG(buffer) {
+  if (buffer.length < 3) return;
+  return buffer[0] === 255 && buffer[1] === 216 && buffer[2] === 255;
+}
+
+function getType(buffer) {
+  if (isBufferJPEG(buffer)) {
+    return JPEG;
+  }
+  if (isBufferPNG(buffer)) {
+    return PNG;
+  }
+}
+
+function mergedTypedArrays(a, b) {
+  var c = new a.constructor(a.length + b.length);
+  c.set(a);
+  c.set(b, a.length);
+  return c;
```

Comment:
so basically this avoid us from referencing Buffer, but giving us buffer compatibility ? or will take in buffer and spit back a typed array?

I can imagine a.constructor to be always the Uint8Array, while the b.constructor can be both Buffer and Uint8Array?

I'm asking because i played very little with buffers in node.

Reply:
I m also wondering about ie11 compatibility here. the `new x.constructor` never works for me in ie11.

Reply:
Ah, you're right about a.constructor always being a Uint8Array. My goal is to maintain whatever type they passed in originally. But I'm remembering now that in node, we wouldn't want to something like `new Buffer()` anyways, since that's deprecated. I'll have to think about this more.

Reply:
if we could accept the fact that we mutate the original Buffer we could just use .set on the original object for the new head.

Do you know if Buffer.slice returns a copy or a reference to the data? if it returns a copy we can just copy the argument that comes and then set the new head over it, and return it.

Did not try it yet, so i m still throwing ideas out

Reply:
It returns a view into the original buffer. Mutating it will mutate the original.

Reply:
Well mutation approch would not work with PNG since the header becomes bigger if the phys chunk is not there.